### DISCUSSION TITLE
How/where does box64 find and use native libraries?

### DISCUSSION THREAD

**Discussion by supsm** [ptitSeb/box64#786] [Q&A] [answered]
I am having a hard time understanding how box64 deals with native libraries. On one hand it seems that box64 will try to load native libraries for *any library*, while on the other hand it seems that box64 only supports wrapped libraries with explicit hardcoded support. If box64 does indeed support arbitrary wrapped libraries, where does it look for these? I have tried changing the `LD_LIBRARY_PATH` (not `BOX64_LD_LIBRARY_PATH` as that seems to be for emulated libraries) to include the directory where the native libs are, but box64 does not seem to attempt loading them.  
I just need this cleared up for now, there is currently nothing specific to my application that I need assistance with. Thank you in advance.

**Comment by ptitSeb** [comment, marked as answer]
Only "hardcoded" libs can be used as native. You can see the list of libs in  the code, look inside "library_list.h"
The libs are open using `dlopen(...)` so LD_LIBRARY_PATH is used indeed.

**Comment by supsm** [comment]
That's unfortunate, I was looking to get cuda working, but it seems like it is too complicated to add support for it. Do you happen to know of any other way I could potentially get this working? It is just a amd64 executable which references cuda.

**Reply by ptitSeb** [reply]
Cuda is certainly a lot of work to wrap, and very niche for now on box64 usage. Just grab the x86_64 version and put it in the program folder, it might just work (non-wrapped library are loaded and emulated, like for the main binary)
